### PR TITLE
Implement ranges::lexicographical_compare

### PR DESCRIPTION
**Pull request by CaseyCarter**
Partially addresses #39.

### CODE REVIEW

File: tests/std/test.lst [hunk 1]
```diff
@@ -262,6 +262,7 @@ tests\P0896R4_ranges_alg_generate_n
 tests\P0896R4_ranges_alg_heap
 tests\P0896R4_ranges_alg_is_permutation
 tests\P0896R4_ranges_alg_is_sorted
+tests\P0896R4_ranges_alg_lexi
```
Comment (StephanTLavavej) [inline tests/std/test.lst:265]: I would like to see the algorithm's name fully spelled out, for ease of searching. We shouldn't be close to the arbitrary path length limit that we recently imposed.

File: tests/std/tests/P0896R4_ranges_alg_lexi/test.cpp [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+using namespace std;
+using P = pair<int, int>;
+
+struct instantiator {
+    static constexpr P left[]          = {{0, 10}, {1, 20}, {2, 30}};
+    static constexpr P right_shorter[] = {{300, 0}, {200, 1}};
+    static constexpr P right_longer[]  = {{300, 0}, {200, 1}, {100, 2}, {0, 3}};
+    static constexpr P right_less[]    = {{300, 0}, {200, 1}, {100, 1}};
+    static constexpr P right_equal[]   = {{300, 0}, {200, 1}, {100, 2}};
+    static constexpr P right_greater[] = {{300, 0}, {200, 1}, {100, 3}};
```
Comment (StephanTLavavej) [inline tests/std/tests/P0896R4_ranges_alg_lexi/test.cpp:21]: Should we have "shorter but greater" and "longer but lesser" test cases?

Comment (StephanTLavavej) [inline tests/std/tests/P0896R4_ranges_alg_lexi/test.cpp:21]: Should we have empty/nonempty, nonempty/empty, empty/empty test cases?

**Comment by StephanTLavavej** [comment]
Thanks for implementing this algorithm! Is this the algorithm in the STL with the greatest number of syllables? :joy_cat:

**Comment by CaseyCarter** [comment]
> Thanks for implementing this algorithm! Is this the algorithm in the STL with the greatest number of syllables? 😹

I believe it's tied for first with `lexicographical_compare_three_way`. (Arbitrarily chosen rules: counting the number of syllables in the fully-qualified name not including `std`.)

**Comment by CaseyCarter** [comment]
> > Thanks for implementing this algorithm! Is this the algorithm in the STL with the greatest number of syllables? 😹
> 
> I believe it's tied for first with `lexicographical_compare_three_way`. (Arbitrarily chosen rules: counting the number of syllables in the fully-qualified name not including `std`.) We really should have a `ranges::lexicographical_compare_three_way`, if for no other reason than to resolve the ambiguity over the longest algorithm name.

**Pull request merged**